hw/top_earlgrey/ip/ast/ast.core

```diff
@@ -8,22 +8,30 @@ filesets:
   files_rtl:
     depend:
       - lowrisc:ip:tlul
+      - lowrisc:prim:all
+      - lowrisc:prim:clock_buf
+      - lowrisc:prim:clock_gating
+      - lowrisc:prim:clock_inv
+      - lowrisc:prim:lc_dec
```

Comment:
Add the 
`- lowrisc:prim:prim_pkg`
dependency here in order to get access to `prim_pkg::ImplXilinx`.

Reply:
'fusesoc is OK with that. but my Verilator local run fail on Impl...

Reply:
I also can not access the CI (private) log to see the fail... If you can, please sent me the log

Reply:
```
Error-[PCTM] Port connection type mismatch
../src/lowrisc_systems_top_earlgrey_asic_0.1/rtl/top_earlgrey_asic.sv, 468
"ast #(top_pkg::ADC_CHANNELS, top_pkg::ADC_DATAW, top_pkg::ENTROPY_STREAM, ast_pkg::Ast2PadOutWidth, ast_pkg::Pad2AstInWidth, ast_pkg::UsbCalibWidth, ) u_ast( .tl_i (base_ast_bus),  .tl_o (ast_base_bus),  .clk_ast_adc_i (1'b0),  .rst_ast_adc_ni (1'b0),  .clk_ast_alert_i (clks_ast.clk_ast_sensor_ctrl_aon_io_div4_secure),  .rst_ast_alert_ni (rsts_ast.rst_ast_sensor_ctrl_aon_sys_io_div4_n[rstmgr_pkg::DomainAonSel]),  .clk_ast_es_i (clks_ast.clk_ast_edn0_main_secure),  .rst_ast_es_ni (rsts_ast.rst_ast_edn0_sys_n[rstmgr_pkg::Domain0Sel]),  .clk_ast_rng_i (clks_ast.clk_ast_entropy_src_main_secure),  .rst_ast_rng_ni (rsts_ast.rst_ast_entropy_src_sys_n[rstmgr_pkg::Domain0Sel]),  .clk_ast_tlul_i (clks_ast.clk_ast_sensor_ctrl_aon_io_div4_secure),  .rst_ast_tlul_ni ... "
  The following expression is illegally connected to port "adc_a0_ai" of 
  module "ast", instance "u_ast". The type of the port does not match that of 
```

And for Verilator: how do you run Verilator? You need to run it through FuseSoC, otherwise the primitive packages won't be correctly generated on the fly.

Reply:
@tjaychen 
I added the ANALOGSIM as a gap stopper so I can check the top_earlgrey_asic for a prepare solution without holding you back. 
As for the struct vs. struct pack. I can not use struct packed with 'real'.

Reply:
@imphil @tjaychen 
I expected this error since the ADC/Analog signals change to a typedef that need to replace the current logic that Tim did on the top_earlgrey_asic. So we only need to finish the integration and all the check will pass,

Reply:
hi Jacob,
if you can't use struct packed, can you do away with the struct altogether?
in our dv environment yesterday, i also tried the following and it worked for DV at least (after various other changes).
What's the reason you have to have a struct?
```
`ifdef ANALOGSIM
typedef real awire_t
`else 
typedef logic awire_t

```

Secondly, for these port level changes, you should also make the related integration changes if CI breaks.

Reply:
Hi Tim,

With struct all the CI tests passed. The only fail I had relate to the missing integration of the analog/AST level in top_earlgrey_asic. I plan to make the changes in top_earlgrey_asic, following your lead. Aslo, @msfschaffner just sent me the missing code for making the "FPGA" code not fail the Verilator.

I will make the changes and re-PR/PR the changed files. There is a good chance that everything will work this time.